modules/data-streams/src/test/java/org/elasticsearch/datastreams/action/GetDataStreamsRequestTests.java

```diff
@@ -8,20 +8,26 @@
  */
 package org.elasticsearch.datastreams.action;
 
+import org.elasticsearch.TransportVersions;
 import org.elasticsearch.action.datastreams.GetDataStreamAction.Request;
 import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.action.support.local.LocalClusterStateHelper;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.common.io.stream.Writeable;
 import org.elasticsearch.test.AbstractWireSerializingTestCase;
 
-public class GetDataStreamsRequestTests extends AbstractWireSerializingTestCase<Request> {
+import java.io.IOException;
+
+public class GetDataStreamsRequestTests extends AbstractWireSerializingTestCase<GetDataStreamsRequestTests.RequestWrapper> {
```

Comment:
I tried something new here. It was bothering me a bit that in the previous PRs where I/we made actions run on the local node, we still needed parts of the (de)serialization of the requests and responses, but because the other half wasn't required anymore, we simply dropped the serialization tests. We still need parts of the (de)serialization for BwC reasons, but I'm not sure if all actions are tested in mixed-cluster/BwC situations and I thus wasn't sure if the serialization code was still being tested. Therefore, I tried to experiment with adding these serialization wrappers. How do we feel about that approach? If we see value in it, I'll backport it to the other actions as well.

Reply:
I don't mind it, but I'm not sure it's worth the extra effort, so I don't think you should necessarily backport it. Do you have an opinion?